core/deployment/src/main/java/io/quarkus/deployment/builditem/AdditionalBootstrapConfigProviderBuildItem.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.deployment.builditem;
+
+import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+/**
+ * Build item to use when an additional {@link ConfigSourceProvider} needs to be
+ * registered to the Bootstrap Config setup.
+ * This is needed because during Bootstrap Config setup, we don't auto-discover providers
+ * but we do want for example the YAML provider to be enabled.
+ */
+public final class AdditionalBootstrapConfigProviderBuildItem extends MultiBuildItem {
```

Comment:
Maybe it's me, but the name is kinda confusing. Maybe something like `AdditionalConfigSourceProviderBuildItem` ?

Reply:
You have a point, I'll make it `AdditionalBootstrapConfigSourceProviderBuildItem`